bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVBridgeHandler.java

```diff
@@ -122,33 +126,52 @@ private void initiateConnexion() {
                 String error = uvResponse.getError();
                 if (error == null) {
                     updateStatus(ThingStatus.ONLINE);
+                    keyVerified = true;
                     return uvResponse.getResult();
                 }
                 throw new OpenUVException(error);
             }
         } catch (JsonSyntaxException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                    String.format("Invalid json received when calling `%s` : %s", url, jsonData));
+            if (jsonData.contains("MongoError")) {
+                statusMessage = String.format("@text/offline.comm-error-faultly-service [ \"%d\" ]",
+                        RECONNECT_DELAY_MIN);
+                scheduleReconnectJob(RECONNECT_DELAY_MIN);
+            } else {
+                statusDetail = ThingStatusDetail.NONE;
+                statusMessage = String.format("@text/offline.invalid-json [ \"%s\", \"%s\" ]", url, jsonData);
```

Comment:
A relatively small message is expected (it will be displayed in UI). Is your `jsonData` small ?
If not, maybe you shoudl add a specific DEBUG log entry to log `jsonData`

Reply:
Size of invalid json is not easy to predict, so I take your proposal.